src/ansi-c/ansi_c_language.cpp

```diff
@@ -135,6 +138,492 @@ bool ansi_c_languaget::generate_support_functions(
     symbol_table, get_message_handler(), object_factory_params);
 }
 
+// Question: Are those [is_*] functions correct? Are they sound/complete?
+bool is_call_to_function(std::string function_name, exprt expr) {
+  if(can_cast_expr<side_effect_expr_function_callt>(expr))
+  {
+    const side_effect_expr_function_callt function_call = to_side_effect_expr_function_call(expr);
+    const exprt function = function_call.function();
+    
+    if(can_cast_expr<symbol_exprt>(function))
+    {
+      const symbol_exprt function_symbol = to_symbol_expr(function);
+      return function_symbol.get_identifier() == function_name;
+    }
+  }
+  return false;
+}
+
+bool is_code_postcondition(exprt expr) {
+  if(can_cast_expr<code_expressiont>(expr))
+  {
+    exprt expr0 = to_code_expression(to_code(expr)).expression();
+    return is_call_to_function("__CPROVER_postcondition", expr0);
+  }
+  return false;
+}
+
+
+exprt::operandst filter_pre_post_conditions(std::string target_function_name, exprt function_body) {
```

Comment:
`target_function_name` should be `irep_idt`, `function_body` should probably be `code_blockt`.